openslides/core/views.py

```diff
@@ -535,22 +534,24 @@ def check_view_permissions(self):
 
     def list(self, request):
         """
-        Lists all config variables. Everybody can see them.
+        Lists all config variables.
         """
-        return Response([{'key': key, 'value': value} for key, value in config.items()])
+        collection = Collection(config.get_collection_string())
+        return Response(collection.as_list_for_user(request.user))
 
     def retrieve(self, request, *args, **kwargs):
         """
-        Retrieves a config variable. Everybody can see it.
+        Retrieves a config variable.
         """
         key = kwargs['pk']
+        collection_element = CollectionElement.from_values(config.get_collection_string(), key)
         try:
-            value = config[key]
-        except ConfigNotFound:
+            content = collection_element.as_dict_for_user(request.user)
+        except ConfigStore.DoesNotExist:
             raise Http404
-        # Attention: The format of this response has to be the same as in
-        # the get_full_data method of ConfigAccessPermissions.
-        return Response({'key': key, 'value': value})
+        if content is None:
```

Comment:
See the next line: when the user has no permission to see the content. See also the comment in utils/access_permissions.py from get_restricted_data()